src/util.c

```diff
@@ -93,9 +93,9 @@ static int checkMC()
             if (dir == NULL) {
                 // No base dir found on any MC, check MC is inserted
                 fd = sysCheckMC();
-                if (fd == 0)
+                if ((fd = 0))
                     mcID = 0x30;
-                else if (fd == 1)
+                else if ((fd = 1))
```

Comment:
This is wrong, you're changing a compare '==' into an assignment '='. The first if will always be 0 now, the second if will aways be 1.

Reply:
Whoops... yea, I didn’t look at that properly.. result of a quick trial and error.. but yea I’m giving fd the value of 0 or 1 there.. sysCheckMCs return values are strange though.. it should work as is at first glance which is why I didn’t bother testing both slots when committing the first time... return 0 if mc0 found return 1 if mc1 found.. but it returns 0 if mc in either slot.. I will change this later